XMonad/Hooks/DynamicLog.hs

```diff
@@ -223,12 +224,24 @@ statusBar :: LayoutClass l Window
                        -- ^ the desired key binding to toggle bar visibility
           -> XConfig l -- ^ the base config
           -> IO (XConfig (ModifiedLayout AvoidStruts l))
-statusBar cmd pp k conf = do
+statusBar cmd pp = statusBar' cmd (return pp)
+
+-- | Like 'statusBar' with the pretty printing options embedded in the
+-- X monad. Useful if printing options need to be modified dynamically.
```

Comment:
Might be useful to clarify that the `X PP` will be re-executed every time the `logHook` runs.

Reply:
@byorgey Added in ba643c5. I'm happy to squash that commit -- I kept it separate only for ease of review.